Mathlib/Data/Finset/Image.lean

```diff
@@ -620,32 +620,89 @@ theorem map_subtype_subset {t : Set α} (s : Finset t) : ↑(s.map (Embedding.su
 
 end Subtype
 
+section Fin
 /-! ### Fin -/
 
+variable {n : ℕ}
+
+/-- Lift a finite set `s : Finset ℕ` to `Finset (Fin n)`,
+given a set in `Fin n` whose image under `Fin.val` is equal to `s`.
+
+This is an auxiliary definition used in the definition of a locally finite order on `Fin n`,
+in order to avoid re-verifying `i < n` for each `i` in an interval. -/
+def finOfImageEq (s : Finset ℕ) (t : Set (Fin n)) (h : Fin.val '' t = s) : Finset (Fin n) := by
```

Comment:
@YaelDillies WDYT about this definition? My main goal is to avoid an extra `Finset.filter` in the computation of `Finset.Icc (_ : Fin _) _`. Another option is to introduce `Finset.attachFin` assuming `s ⊆ .range n` instead.

Reply:
Do you want to use `Finset.filterMap` maybe?

Reply:
No, I don't want to verify that each number is less than `n`.

Reply:
Changed to `attachFin`.

Reply:
... done. CC: @YaelDillies